operators/src/processing/temporal_raster_aggregation.rs

```diff
@@ -141,85 +160,113 @@ where
+                    no_data_ignoring_fold_future::<P, MinIgnoreNoDataAccFunction>,
+                    P::max_value(),
+                )
+                .into_raster_overlap_adapter(&self.source, query, ctx, self.tiling_specification)
+                .boxed()),
+            Aggregation::Min {
+                ignore_no_data: false,
+            } => Ok(self
+                .create_subquery(fold_future::<P, MinAccFunction>, P::max_value())
+                .into_raster_overlap_adapter(&self.source, query, ctx, self.tiling_specification)
+                .boxed()),
+            Aggregation::Max {
+                ignore_no_data: true,
+            } => Ok(self
+                .create_subquery(
+                    no_data_ignoring_fold_future::<P, MaxIgnoreNoDataAccFunction>,
+                    P::min_value(),
+                )
+                .into_raster_overlap_adapter(&self.source, query, ctx, self.tiling_specification)
+                .boxed()),
+            Aggregation::Max {
+                ignore_no_data: false,
+            } => Ok(self
+                .create_subquery(fold_future::<P, MaxAccFunction>, P::min_value())
+                .into_raster_overlap_adapter(&self.source, query, ctx, self.tiling_specification)
+                .boxed()),
+            Aggregation::First {
+                ignore_no_data: true,
+            } => {
+                let no_data_value = self
+                    .no_data_value
+                    .ok_or(error::Error::TemporalRasterAggregationFirstValidRequiresNoData)?;
+                Ok(self
+                    .create_subquery(
+                        no_data_ignoring_fold_future::<P, FirstValidAccFunction>,
+                        no_data_value,
+                    )
+                    .into_raster_overlap_adapter(
                         &self.source,
                         query,
-                        self.tiling_specification,
                         ctx,
-                        spec,
-                    );
-
-                    Ok(s.boxed())
-                } else {
-                    let spec = TemporalRasterAggregationSubQuery {
-                        fold_fn: fold_future::<P, MinAccFunction>,
-                        no_data_value: self.no_data_value,
-                        initial_value: P::max_value(),
-                        step: self.window,
-                    };
-
-                    let s = RasterOverlapAdapter::<'a, P, _, _>::new(
+                        self.tiling_specification,
+                    )
+                    .boxed())
+            }
+            Aggregation::First {
+                ignore_no_data: false,
+            } => {
+                let no_data_value = self
+                    .no_data_value
+                    .ok_or(error::Error::TemporalRasterAggregationFirstValidRequiresNoData)?;
+                Ok(self
+                    .create_subquery(first_tile_fold_future::<P>, no_data_value)
```

Comment:
can we avoid processing all tiles of the timeseries when all we want is the first one?

Reply:
i don't think that is possible here. you need to cancel the subquery for that.